satellite/overlay/selection_test.go

```diff
@@ -301,14 +301,61 @@ func TestNodeSelectionGracefulExit(t *testing.T) {
 	})
 }
 
+func TestFindStorageNodesDistinctIPs(t *testing.T) {
+	if runtime.GOOS == "darwin" {
+		t.Skip("Test does not work with macOS")
+	}
+	testplanet.Run(t, testplanet.Config{
+		SatelliteCount: 1, StorageNodeCount: 5, UplinkCount: 1,
+		Reconfigure: testplanet.Reconfigure{
+			// will create 3 storage nodes with same IP; 2 will have unique
+			UniqueIPCount: 2,
+		},
+	}, func(t *testing.T, ctx *testcontext.Context, planet *testplanet.Planet) {
+		satellite := planet.Satellites[0]
+
+		var excludedNodes storj.NodeIDList
+		addrs := make(map[storj.NodeID]struct{})
+		for _, node := range planet.StorageNodes {
+			if _, ok := addrs[node.ID()]; ok {
+				addrs[node.ID()] = struct{}{}
+			} else {
+				// Add one ID to the excluded nodes list.
```

Comment:
Specifically, you are trying to add one of the nodes that *has* a duplicate IP to the list, correct? Like if you have nodes 1-5, and 1-3 have IP1, 4 has IP2, and 5 has IP3, you are trying to add node 1, 2, or 3 to `excludedNodes`?
If yes, I think we should clarify this as part of this comment.

Reply:
yes that's correct, i'll add to comment